File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/permission/repository/PermissionRepository.java
package com.nivasafinance.features.rolemanagement.permission.repository;

import com.nivasafinance.features.rolemanagement.permission.entity.Permission;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PermissionRepository extends JpaRepository<Permission, Long> {
    List<Permission> findByIdIn(List<Long> ids);
    
    Page<Permission> findAll(Pageable pageable);
    Page<Permission> findByNameContainingIgnoreCase(String name, Pageable pageable);
}

File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/permission/repository/PermissionRepository.kt
package com.nivasafinance.features.rolemanagement.permission.repository

import com.nivasafinance.features.rolemanagement.permission.entity.Permission
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface PermissionRepository : JpaRepository<Permission, UUID> {
    fun findByIdIn(ids: List<UUID>): List<Permission>
}
